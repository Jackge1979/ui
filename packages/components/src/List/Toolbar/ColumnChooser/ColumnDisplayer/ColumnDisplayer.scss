$tc-border-color: #ededed !default;
$tc-icon-size: 1.3rem !default;

.tc-column-displayer {
	display: flex;
	position: relative;
	align-items: center;
	height: 4rem;
	min-height: 4rem;
	&-dragging, 
	&:hover {
		background-color: $wild-sand;
	}
	&-dragging {
		position: absolute;
		content: '';
		top: 0;
		left: 0;
		bottom: 0;
		right: 0;
		z-index: -1;
		margin-right: -$padding-large;
		margin-left: -$padding-large;
	}

	&-visibility {
		align-items: center;
		border-right: 0.1rem solid $tc-border-color;
		display: flex;
		height: 70%;
		padding-right: $padding-small;
		&-checkbox {
			margin: 0;
			position: relative;
		}
		:global(.tc-svg-icon) {
			width: $tc-icon-size;
			height: $tc-icon-size;
		}
	}
	&-label {
		display: flex;
		padding-left: $padding-large;
		width: 100%;
	}
	&-order {
		align-items: center;
		display: flex;
		justify-content: flex-end;
		width: 100%;
		&-value,
		&-input-text {
			font-weight: bold;
			font-size: 1.5rem;
			padding-right: $padding-smaller;
		}
		&-input-text {
			text-align: right;
			max-width: 5rem;
		}
		:global(.btn) {
			color: $dove-gray;
			padding: 0;
		}
	}
}
