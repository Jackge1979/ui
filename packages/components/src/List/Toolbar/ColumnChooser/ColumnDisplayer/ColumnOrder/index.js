import Component from './ColumnOrder.component';

export default Component;
