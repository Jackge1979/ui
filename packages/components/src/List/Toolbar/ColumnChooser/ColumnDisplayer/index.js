import Component from './ColumnDisplayer.component';

export default Component;
