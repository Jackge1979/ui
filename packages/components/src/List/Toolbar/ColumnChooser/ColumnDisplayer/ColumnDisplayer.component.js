import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Icon from '../../../../Icon';
import ColumnOrder from './ColumnOrder';
import { columnChooserContext } from '../Content/columnChooser.context';

import theme from './ColumnDisplayer.scss';

export const ColumnLabel = props => {
	return (
		<span className={classNames(theme['tc-column-displayer-label'], 'tc-column-displayer-label')}>
			{props.label}
		</span>
	);
};

export const ColumnVisibility = ({ index, locked, value }) => {
	const { onChangeVisibility, t } = useContext(columnChooserContext);
	return (
		<div
			className={classNames(
				theme['tc-column-displayer-visibility'],
				'tc-column-displayer-visibility',
			)}
		>
			{locked ? (
				<Icon name="talend-locked" />
			) : (
				<input
					aria-label={t('CHECKBOX_VISIBILITY_COLUMN_CHOOSER', {
						defaultValue: 'change visibility',
					})}
					className={classNames(
						theme['tc-column-displayer-visibility-checkbox'],
						'tc-column-displayer-visibility-checkbox',
					)}
					onChange={event => onChangeVisibility(index)(event, !value)}
					type="checkbox"
					checked={!value}
					value={!value}
				/>
			)}
		</div>
	);
};

ColumnVisibility.propTypes = {
	locked: PropTypes.bool,
	value: PropTypes.bool,
};

const ColumnDisplayer = ({ label, hidden, locked, order, length, children, index, ...rest }) => {
	const { id } = useContext(columnChooserContext);
	if (rest.default) {
		return (
			<div
				id={`${id}-displayer`}
				className={classNames(theme['tc-column-displayer'], 'tc-column-displayer')}
			>
				<ColumnDisplayer.ColumnVisibility index={index} value={hidden} locked={locked} />
				<ColumnDisplayer.ColumnLabel label={label} />
				<ColumnDisplayer.ColumnOrder index={index} length={length} locked={locked} value={order} />
			</div>
		);
	}
	return (
		<div
			id={`${id}-displayer`}
			className={classNames(theme['tc-column-displayer'], 'tc-column-displayer')}
		>
			{children}
		</div>
	);
};

ColumnDisplayer.ColumnVisibility = ColumnVisibility;
ColumnDisplayer.ColumnLabel = ColumnLabel;
ColumnDisplayer.ColumnOrder = ColumnOrder;

ColumnDisplayer.propTypes = {
	hidden: PropTypes.bool,
	label: PropTypes.string.isRequired,
	length: PropTypes.number.isRequired,
	locked: PropTypes.bool,
	onBlurOrder: PropTypes.func.isRequired,
	onChangeVisibility: PropTypes.func.isRequired,
	onKeyPressOrder: PropTypes.func.isRequired,
	order: PropTypes.number.isRequired,
	t: PropTypes.func.isRequired,
};

export default ColumnDisplayer;
