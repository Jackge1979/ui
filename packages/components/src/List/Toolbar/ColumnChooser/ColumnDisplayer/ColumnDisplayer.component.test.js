import React from 'react';
import { shallow } from 'enzyme';
import ColumnDisplayer, {
	ColumnOrder,
	OrderDisplay,
	ColumnVisibility,
} from './ColumnDisplayer.component';

describe('ColumnVisibility', () => {
	const onChange = jest.fn();
	const value = true;
	it('should render when locked', () => {
		// given
		const props = {
			locked: true,
			onChange,
			value,
		};
		// when
		const wrapper = shallow(<ColumnVisibility {...props} />);
		// then
		expect(wrapper.getElement()).toMatchSnapshot();
	});
	it('should render the checkbox', () => {
		// given
		const props = {
			locked: false,
			onChange,
			value,
		};
		// when
		const wrapper = shallow(<ColumnVisibility {...props} />);
		// then
		expect(wrapper.getElement()).toMatchSnapshot();
	});
});

describe('OrderDisplay', () => {
	it('should return', () => {
		// given
		const props = {
			order: 1,
			length: 10,
		};
		// when
		const wrapper = shallow(<OrderDisplay {...props} />);
		// then
		expect(wrapper.getElement()).toMatchSnapshot();
	});
});
