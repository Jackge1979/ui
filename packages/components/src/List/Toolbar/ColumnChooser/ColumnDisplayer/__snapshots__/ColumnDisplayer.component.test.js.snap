// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColumnVisibility should render the checkbox 1`] = `
<input
  checked={false}
  className="theme-tc-column-displayer-visibility-checkbox tc-column-displayer-visibility-checkbox"
  onChange={[Function]}
  type="checkbox"
  value={false}
/>
`;

exports[`ColumnVisibility should render when locked 1`] = `
<Icon
  name="talend-locked"
/>
`;

exports[`OrderDisplay should return 1`] = `
<React.Fragment>
  <span
    className="theme-tc-column-displayer-order-value tc-column-displayer-order-value"
  >
    1
  </span>
  / 10
</React.Fragment>
`;
